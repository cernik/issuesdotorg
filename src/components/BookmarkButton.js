// @flow
import React from 'react';
import {TouchableOpacity} from 'react-native';

import {noop} from '../utils/constants';

import BookmarkIcon from '../assets/svg/bookmark.svg';
import BookmarkAddIcon from '../assets/svg/bookmark-add.svg';

type BookmarkButtonProps = {
  selected?: boolean,
  onPress?: () => void,
  style?: Object,
};

const BookmarkButton = ({
  selected = false,
  onPress = noop,
  style,
}: BookmarkButtonProps) => (
  <TouchableOpacity onPress={onPress} style={style}>
    {selected ? <BookmarkIcon /> : <BookmarkAddIcon />}
  </TouchableOpacity>
);

export default BookmarkButton;
