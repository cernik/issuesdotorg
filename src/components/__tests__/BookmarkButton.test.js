// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import BookmarkButton from '../BookmarkButton';

describe('snapshot', () => {
  it('selected version renders correctly', () => {
    const tree = renderer.create(<BookmarkButton selected />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('non-selected version renders correctly', () => {
    const tree = renderer.create(<BookmarkButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('accepts onPress function', () => {
    const tree = renderer
      .create(<BookmarkButton onPress={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
