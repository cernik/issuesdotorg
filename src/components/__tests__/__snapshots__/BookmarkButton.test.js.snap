// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot accepts onPress function 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <SvgMock />
</View>
`;

exports[`snapshot non-selected version renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <SvgMock />
</View>
`;

exports[`snapshot selected version renders correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <SvgMock />
</View>
`;
